commands/utils/params.go

```diff
@@ -11,159 +17,163 @@ import (
 	url := strings.TrimSuffix(getTrimmedEnv(JFrogUrlEnv), "/")
 	xrUrl := strings.TrimSuffix(getTrimmedEnv(jfrogXrayUrlEnv), "/")
 	rtUrl := strings.TrimSuffix(getTrimmedEnv(jfrogArtifactoryUrlEnv), "/")
 	if xrUrl != "" && rtUrl != "" {
-		params.Server.XrayUrl = xrUrl + "/"
-		params.Server.ArtifactoryUrl = rtUrl + "/"
+		server.XrayUrl = xrUrl + "/"
+		server.ArtifactoryUrl = rtUrl + "/"
 	} else {
 		if url == "" {
-			return fmt.Errorf("%s or %s and %s environment variables are missing", JFrogUrlEnv, jfrogXrayUrlEnv, jfrogArtifactoryUrlEnv)
+			return server, fmt.Errorf("%s or %s and %s environment variables are missing", JFrogUrlEnv, jfrogXrayUrlEnv, jfrogArtifactoryUrlEnv)
 		}
-		params.Server.Url = url + "/"
-		params.Server.XrayUrl = url + "/xray/"
-		params.Server.ArtifactoryUrl = url + "/artifactory/"
+		server.Url = url + "/"
+		server.XrayUrl = url + "/xray/"
+		server.ArtifactoryUrl = url + "/artifactory/"
 	}
 
 	password := getTrimmedEnv(JFrogPasswordEnv)
 	user := getTrimmedEnv(JFrogUserEnv)
 	if password != "" && user != "" {
-		params.Server.User = user
-		params.Server.Password = password
+		server.User = user
+		server.Password = password
 	} else if accessToken := getTrimmedEnv(JFrogTokenEnv); accessToken != "" {
-		params.Server.AccessToken = accessToken
+		server.AccessToken = accessToken
 	} else {
-		return fmt.Errorf("%s and %s or %s environment variables are missing", JFrogUserEnv, JFrogPasswordEnv, JFrogTokenEnv)
+		return coreconfig.ServerDetails{}, fmt.Errorf("%s and %s or %s environment variables are missing", JFrogUserEnv, JFrogPasswordEnv, JFrogTokenEnv)
 	}
-	// Non-mandatory Xray context params
-	_ = readParamFromEnv(jfrogWatchesEnv, &params.Watches)
-	_ = readParamFromEnv(jfrogProjectEnv, &params.Project)
-	return nil
+	return server, nil
 }
 
-func extractGitParamsFromEnv(params *FrogbotParams) error {
+func extractGitParamsFromEnv() (GitParams, error) {
 	var err error
-
+	gitParams := GitParams{}
 	// Non-mandatory Git Api Endpoint
-	_ = readParamFromEnv(GitApiEndpointEnv, &params.ApiEndpoint)
+	_ = readParamFromEnv(GitApiEndpointEnv, &gitParams.ApiEndpoint)
 
-	if params.GitProvider, err = extractVcsProviderFromEnv(); err != nil {
-		return err
-	}
-	if err = readParamFromEnv(GitRepoOwnerEnv, &params.RepoOwner); err != nil {
-		return err
+	if gitParams.GitProvider, err = extractVcsProviderFromEnv(); err != nil {
+		return GitParams{}, err
 	}
-	if err = readParamFromEnv(GitRepoEnv, &params.Repo); err != nil {
-		return err
+	if err = readParamFromEnv(GitRepoOwnerEnv, &gitParams.RepoOwner); err != nil {
+		return GitParams{}, err
 	}
-	if err = readParamFromEnv(GitTokenEnv, &params.Token); err != nil {
-		return err
+
+	if err = readParamFromEnv(GitTokenEnv, &gitParams.Token); err != nil {
+		return GitParams{}, err
 	}
 	// Non-mandatory git branch and pr id.
-	_ = readParamFromEnv(GitBaseBranchEnv, &params.BaseBranch)
+	_ = readParamFromEnv(GitBaseBranchEnv, &gitParams.BaseBranch)
```

Comment:
I'm not sure what you mean because we use it for other mandatory environment variables and if not configured, an error is returned from 'readParamFromEnv' and the process eventually stops.